src/libraries/System.Private.CoreLib/src/System/IO/Path.Unix.cs

```diff
@@ -58,9 +40,32 @@ public static string GetFullPath(string path, string basePath)
                 throw new ArgumentException(SR.Argument_InvalidPathChars);
 
             if (IsPathFullyQualified(path))
-                return GetFullPath(path);
+                return GetPartiallyQualifiedPath(path, true);
```

Comment:
There is a check `\0` a few lines before this line. It should not be necessary to redo the check.